quartz/plugins/transformers/ofm.ts

```diff
@@ -346,11 +352,32 @@ export const ObsidianFlavoredMarkdown: QuartzTransformerPlugin<Partial<Options>
               }
             })
           }
-
           mdastFindReplace(tree, replacements)
         }
       })
 
+      //go through mdast, find img nodes with video extension, convert them to html nodes
+      if (opts.enableVideoEmbed) {
+        plugins.push(() => {
+          return (tree: Root, _file) => {
+            visit(tree, "image", (node, index, parent) => {
+              const match = node.url.match(videoExtensionRegex)
+              if (match && parent) {
```

Comment:
```suggestion
              if (parent && match) {
```

always do the cheaper conditional check first so short-circuiting is faster